### PR TITLE
Add clarification to MSON element attributes

### PR DESCRIPTION
**Pull request by zdne**
Clarify types in `content` and of `sample` and `default` attributes.

Add explanation of `typeAttributes` as well as `variable` attribute.

### CODE REVIEW

**Comment by zdne** [comment]
This PR comes from @klokane 's request for clarification. 

Per [MSON spec](https://github.com/apiaryio/mson/blob/master/MSON%20Specification.md#45-default) there can be at maximum only one default value at any given moment. However there MAY be multiple sample values. 

When it comes to `variable` attribute – in most cases this one is used instead of the "sample" `typeAttributes` value to signalize a value specified is sample and not concrete value.

For example:

``` markdown
- color: red (sample)
```

is equivalent to 

``` markdown
- color: *red*
```

However it may be also used to indicate variable property name:

``` markdown
- *color*
```

which basically says the "color" is just a variable name for a property but its value may be different e.g.:

``` json
{ "red": null }
```

or 

``` json
{ "blue": null }
```

Finally it may denote a variable type name (think of it as C++ `typename` keyword) 

``` markdown
# My Object (*T*)
```

**Comment by klokane** [comment]
Thanks for explain :+1

File: namespaces/mson-namespace.md [hunk 1]
```diff
@@ -72,7 +72,7 @@ Note: Not every MSON _Base Type_ is presented in Refract primitive types and vic
 
 # MSON DOM Elements
 
-## MSON Element (Element)
+## MSON Element (Element[*T*])
```
Comment (smizell) [inline namespaces/mson-namespace.md:75]: What does this change mean?

File: namespaces/mson-namespace.md [hunk 2]
```diff
@@ -85,22 +85,28 @@ Note: In MSON DOM _Nested Member Types_ _Type Section_ is the `content` of the e
...
+            - optional - This element is optional in parent's content
+            - fixed - The `content` value is immutable.
+            - sample - The `content` value is a sample value.
+            - default - The `content` value is a default value.
 
-		The type of of `sample` attribute MUST match the type of element's `content`.
+    - `variable` (boolean) - The `content` value is a _Variable Value_
 
-    - `default` - Default value for _Member Types_
+      The `content` value is either a _Variable Type Name_, _Variable Value_ or _Variable Property Name_.
```
Comment (smizell) [inline namespaces/mson-namespace.md:96]: You may need a little clarity here. Does this mean:
- If this is true, "The `content` value is either a _Variable Type Name_, _Variable Value_ or _Variable Property Name_."
- "The `content` value is either a _Variable Type Name_, _Variable Value_ or _Variable Property Name_," and if this is true, the content value is _Variable Type Name_

I think you can read it both ways here unless you put it together with the description above it.

**Comment by smizell** [comment]
@zdne A couple of comments. Looks good once you check those out.

**Comment by smizell** [comment]
@zdne Ping! Just seeing if you wanted to look at this.

**Comment by zdne** [comment]
@smizell oh sorry will try to updated this ASAP lats few days were :meat_on_bone:

File: namespaces/mson-namespace.md (hunk 1 above)
Comment (zdne) [inline namespaces/mson-namespace.md:75]: This was actually invalid according to current MSON syntax

Comment (smizell) [inline namespaces/mson-namespace.md:75]: This being the original, or the change? I'm assuming the original.

**Comment by zdne** [comment]
@smizell udpated

**Pull request merged**